src/contracts/libraries/BN254.sol

```diff
@@ -131,6 +131,29 @@ library BN254 {
         require(success, "ec-add-failed");
     }
 
+    /**
+     * @notice an optimized ecMul implementation that takes log_2(s) ecAdds
+     * @param p the point to multiply
+     * @param s the scalar to multiply by
+     * @dev this function is only safe to use if the scalar is 9 bits or less
+     */ 
+    function scalar_mul_tiny(BN254.G1Point memory p, uint16 s) internal view returns (BN254.G1Point memory) {
+        // the accumulated product to return
+        BN254.G1Point memory acc = BN254.G1Point(0, 0);
+        // the 2^n*p to add to the accumulated product in each iteration
+        BN254.G1Point memory p2n = p;
+        // loop through each bit of s
+        for (uint8 i = 0; i < 9; i++) {
+            // if the bit is 1, add the 2^n*p to the accumulated product
+            if (s >> i & 1 == 1) {
+                acc = plus(acc, p2n);
+            }
+            // double the 2^n*p for the next iteration
+            p2n = plus(p2n, p2n);
+        }
+        return acc;
+    }
+
```

Comment:
what's the usage for this?
Should it explicitly enforce `s < (2**10)?

Reply:
its a way to add without calling the ecMul precompile. @dev is a precondition and doesn't require to save gas. not sure whether this is actually more efficient because it needs the extra call gas to the precompile, but its a start.

Reply:
OK, gotcha. IMO add the require statement for safety. The tiny bit of extra gas seems worth it to make the library more idiot-proof

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/a4890ff2b3a0bedffa1bf33d71c8006aa97ea659